File: test_transcript_client.js
/**
 * Simple Node.js/JavaScript example for testing the transcript WebSocket server
 * 
 * Usage:
 *   node test_transcript_client.js
 *   node test_transcript_client.js --host localhost --port 8765
 */

const WebSocket = require('ws');

// Parse command line arguments
const args = process.argv.slice(2);
let host = 'localhost';
let port = 8765;

for (let i = 0; i < args.length; i++) {
    if (args[i] === '--host' && args[i + 1]) {
        host = args[i + 1];
        i++;
    } else if (args[i] === '--port' && args[i + 1]) {
        port = parseInt(args[i + 1]);
        i++;
    }
}

const wsUrl = `ws://${host}:${port}`;

console.log('='.repeat(60));
console.log('🧪 Testing Transcript WebSocket Server');
console.log('='.repeat(60));
console.log(`Connecting to: ${wsUrl}`);
console.log('Press Ctrl+C to disconnect');
console.log('-'.repeat(60));

let transcriptCount = 0;
let startTime = null;

const ws = new WebSocket(wsUrl);

ws.on('open', () => {
    console.log('\n✅ Connected to server!\n');
    console.log('🎙️  Listening for transcripts...\n');
    console.log('-'.repeat(60));
    startTime = Date.now();
});

ws.on('message', (data) => {
    try {
        const message = JSON.parse(data.toString());
        
        if (message.type === 'connected') {
            console.log(`📨 Server: ${message.message}\n`);
        } else if (message.type === 'transcript') {
            transcriptCount++;
            const startTime = message.start_time || 0;
            const endTime = message.end_time || 0;
            const text = message.text || '';
            
            console.log(`[${startTime.toFixed(2)}s - ${endTime.toFixed(2)}s]: ${text}`);
        } else if (message.type === 'pong') {
            // Server responded to ping
            console.log('✓ Ping received');
        }
    } catch (error) {
        console.error('Error parsing message:', error);
    }
});

ws.on('error', (error) => {
    console.error('\n❌ WebSocket error:', error.message);
    console.log('\nMake sure the server is running:');
    console.log(`  python transcript_server.py --port ${port}`);
    process.exit(1);
});

ws.on('close', () => {
    const elapsed = startTime ? Math.floor((Date.now() - startTime) / 1000) : 0;
    console.log('\n' + '-'.repeat(60));
    console.log(`\n✅ Test complete!`);
    console.log(`   Transcripts received: ${transcriptCount}`);
    console.log(`   Connected time: ${elapsed}s`);
    console.log('='.repeat(60));
    process.exit(0);
});

// Send periodic ping to keep connection alive
const pingInterval = setInterval(() => {
    if (ws.readyState === WebSocket.OPEN) {
        ws.send(JSON.stringify({ type: 'ping' }));
    }
}, 30000); // Every 30 seconds

// Handle Ctrl+C gracefully
process.on('SIGINT', () => {
    console.log('\n\nDisconnecting...');
    clearInterval(pingInterval);
    if (ws) {
        ws.close();
    }
});

File: test_transcript_client.py
#!/usr/bin/env python3
"""
Simple test client for the transcript WebSocket server
This allows you to test the server before integrating with Electron
"""
import asyncio
import websockets
import json
import sys
import signal

running = True

def signal_handler(sig, frame):
    """Handle Ctrl+C gracefully"""
    global running
    print("\n\nDisconnecting...")
    running = False

async def test_client(uri):
    """Test client that connects to the transcript server"""
    global running
    
    print("=" * 60)
    print("🧪 Testing Transcript WebSocket Server")
    print("=" * 60)
    print(f"Connecting to: {uri}")
    print("Press Ctrl+C to disconnect")
    print("-" * 60)
    
    try:
        async with websockets.connect(uri) as websocket:
            print("✅ Connected to server!\n")
            
            # Wait for initial connection message
            try:
                message = await asyncio.wait_for(websocket.recv(), timeout=2.0)
                data = json.loads(message)
                if data.get('type') == 'connected':
                    print(f"📨 Server: {data.get('message', '')}\n")
            except asyncio.TimeoutError:
                pass
            
            print("🎙️  Listening for transcripts...\n")
            print("-" * 60)
            
            # Listen for transcripts
            while running:
                try:
                    message = await asyncio.wait_for(websocket.recv(), timeout=1.0)
                    data = json.loads(message)
                    
                    if data.get('type') == 'transcript':
                        start_time = data.get('start_time', 0)
                        end_time = data.get('end_time', 0)
                        text = data.get('text', '')
                        
                        print(f"[{start_time:.2f}s - {end_time:.2f}s]: {text}")
                    
                    elif data.get('type') == 'pong':
                        # Server responded to ping
                        pass
                        
                except asyncio.TimeoutError:
                    # Send periodic ping to keep connection alive
                    try:
                        await websocket.send(json.dumps({'type': 'ping'}))
                    except:
                        break
                    continue
                except websockets.exceptions.ConnectionClosed:
                    print("\n❌ Connection closed by server")
                    break
                    
    except ConnectionRefusedError:
        print("❌ Connection refused!")
        print("\nMake sure the server is running:")
        print("  python transcript_server.py --port 8765")
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)
    
    print("\n✅ Test complete!")
    print("=" * 60)

def main():
    import argparse
    
    parser = argparse.ArgumentParser(
        description="Test client for transcript WebSocket server",
        formatter_class=argparse.RawDescriptionHelpFormatter,
        epilog="""
Examples:
  # Connect to default server
  python test_transcript_client.py
  
  # Connect to custom port
  python test_transcript_client.py --port 8765
  
  # Connect to remote server
  python test_transcript_client.py --host 192.168.1.100 --port 8765
        """
    )
    
    parser.add_argument(
        "--host",
        type=str,
        default="localhost",
        help="Server host (default: localhost)"
    )
    
    parser.add_argument(
        "--port",
        type=int,
        default=8765,
        help="Server port (default: 8765)"
    )
    
    args = parser.parse_args()
    
    # Set up signal handler
    signal.signal(signal.SIGINT, signal_handler)
    
    uri = f"ws://{args.host}:{args.port}"
    
    try:
        asyncio.run(test_client(uri))
    except KeyboardInterrupt:
        print("\n\nDisconnected.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()



